### PR TITLE
Add tubes CFD example

### PR DESCRIPTION
**Pull request by akaszynski**
There are a ton of awesome projects and datasets at https://www.simscale.com/ and this PR adds one of the examples to our Gallery and downloadable datasets.

```py
>>> import pyvista as pv
>>> from pyvista import examples
>>> dataset = examples.download_openfoam_tubes()
>>> y_slice = dataset.slice('y')
>>> pl = pv.Plotter()
>>> _ = pl.add_mesh(
...     y_slice,
...     scalars='U',
...     lighting=False,
...     scalar_bar_args={'title': 'Flow Velocity'},
... )
>>> _ = pl.add_mesh(dataset, color='w', opacity=0.25)
>>> pl.enable_anti_aliasing()
>>> pl.show()
```

flow

### CODE REVIEW

**Review by tkoyama010 — APPROVED** [review]

**Comment by tkoyama010** [comment]
Oh. It's nice! This service seems to be similar to onescale.
image

**Comment by MatthewFlamm** [comment]
This is now all too late, but I suggest not using `interpolate` when using data from a mesh with cells from CFD.  It is better to use sample or probe (my understanding is that these are mostly the same but they need to be aligned with each other).  This is because these methods find the cell with the point to probe, which is closer to how the FVM approach works.

**Comment by tkoyama010** [comment]
I'm sorry. We should have waited for you, the CFD person. If you send us a pull request for changes, we will review it again.

**Comment by akaszynski** [comment]
@MatthewFlamm, I’ll open a new PR with your suggestions. Also, please see #4235

File: pyvista/examples/downloads.py [hunk 1]
```diff
@@ -3904,6 +3904,60 @@ def download_cavity(load=True):  # pragma: no cover
...
+    """
+    filename = _download_archive(
+        'fea/turbo_incompressible/Turbo-Incompressible_3-Run_1-SOLUTION_FIELDS.zip',
+        target_file='case.foam',
+    )
+    if not load:
+        return filename
+    reader = pyvista.OpenFOAMReader(filename)
+    reader.set_active_time_value(1000)
+    return reader.read()[0]
```
[the remark this answers is not in the document]
Comment (akaszynski) [reply pyvista/examples/downloads.py:3958]: Resolved in #4235.

File: examples/99-advanced/openfoam-tubes.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+
+###############################################################################
+# Plot Steamlines - Flow Velocity
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Generate streamlines using :func:`streamlines() <pyvista.DataSetFilters.streamlines>`.
+
+lines, src = mesh.streamlines(
+    vectors='U',
+    source_center=(0, 0, 0),
```
[the remark this answers is not in the document]
Comment (akaszynski) [reply examples/99-advanced/openfoam-tubes.py:47]: Resolved in #4235.

**Pull request merged**